app/services/payment_provider_customers/update_service.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module PaymentProviderCustomers
+  class UpdateService < BaseService
+    def initialize(customer)
+      @customer = customer
+
+      super(nil)
+    end
+
+    def call
+      result = "PaymentProviderCustomers::#{provider_name}Service".constantize.new(customer.provider_customer).update
```

Comment:
What about using the factory design pattern here rather than meta-prog?

We already have a `PaymentProviderCustomers::Factory` that can be used as an example.

The reason behind this is that meta prog tends to make code maintenance harder on the long run.